pennylane/ops/functions/equal.py

```diff
@@ -234,6 +234,9 @@ def _equal_operators(
     ):  # clarifies cases involving PauliX/Y/Z (Observable/Operation)
         return False
 
+    if isinstance(op1, qml.Identity):
+        return True
```

Comment:
This seems quite drastic, what is the logic behind this change? 😅 

Wouldnt this make
```
op1 = qml.s_prod(1., qml.Identity())
op2 = qml.s_prod(1., qml.X(0))

qml.equal(op1, op2) == True
```
?

Reply:
No, because there is a condition before the one I added that asserts that both operators are of the same type. The only impact of the changes made to this file are that Identities will always be equivalent regardless of wires.

Reply:
Ah nice! Maybe leave a developer comment to make that clear